govc/device/clock/add.go

```diff
@@ -0,0 +1,91 @@
+/*
+Copyright (c) 2022 VMware, Inc. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package floppy
+
+import (
+	"context"
+	"flag"
+	"fmt"
+
+	"github.com/vmware/govmomi/govc/cli"
+	"github.com/vmware/govmomi/govc/flags"
+	"github.com/vmware/govmomi/vim25/types"
+)
+
+type add struct {
+	*flags.VirtualMachineFlag
+}
+
+func init() {
+	cli.Register("device.clock.add", &add{})
+}
+
+func (cmd *add) Register(ctx context.Context, f *flag.FlagSet) {
+	cmd.VirtualMachineFlag, ctx = flags.NewVirtualMachineFlag(ctx)
+	cmd.VirtualMachineFlag.Register(ctx, f)
+}
+
+func (cmd *add) Description() string {
+	return `Add precision clock device to VM.
+
+Examples:
+  govc device.clock.add -vm $vm
+  govc device.info clock-*`
+}
+
+func (cmd *add) Run(ctx context.Context, f *flag.FlagSet) error {
+	vm, err := cmd.VirtualMachine()
+	if err != nil {
+		return err
+	}
+
+	if vm == nil {
```

Comment:
question: is this check needed, i.e. there is a case where err==nil but no vm returned?

Reply:
Yes, see: https://github.com/vmware/govmomi/blob/be5528b51ec58f28601350e466e662727cf6c7c4/govc/flags/virtual_machine.go#L102

Although, I'm not sure why we did that instead of returning `ErrHelp`. There might have been other cases in the past where a vm flag was optional, but I only see 1 here: https://github.com/vmware/govmomi/blob/be5528b51ec58f28601350e466e662727cf6c7c4/govc/vm/option/info.go#L103

Something we could refactor in another PR.